### DISCUSSION TITLE
How does the DNS emulator work?

### DISCUSSION THREAD

**Discussion by GrinchERR** [3kh0/ext-remover#506] [Help JOIN TITANIUM NETWORK TO SHARE PLEASE] [answered]
I don't understand how to use the DNS emulator or what it does. PLZ HELP

**Comment by Aka-but-nice** [comment, marked as answer]
What it does is if you have securly, it blocks all requests from Securly, in return, securly can't block your sites. This is recommended if you're in school, it emulates a DNS Which blocks securly.com

**Reply by Aka-but-nice** [reply]
If you have a different blocker, I could make a dns emulator for that specific blocker.

**Reply by Darcii** [reply]
> If you have a different blocker, I could make a dns emulator for that specific blocker.

I have goguardian as our schools main blocker but, a secondary blocker called Cisco Umbrella which blocks the REALLY good sites like soapy2day discord etc. Can you make a emulator for that?

**Reply by Darcii** [reply]
Oh i forgot to mention im past gg

**Reply by amelia-bartlett** [reply]
OHHH THIS IS SO GOOD THANKS FOR TELLING ME ABOUT THIS PROGRAM

I actually reverse engineered the MS Edge extension of Securly and figured out this was possible (I first implemented it with Burp Suite and Proxy Action Rules)

Also, its not a DNS emulator, its a DNS server (you could probably also set up your own with pi-hole)

**Reply by amelia-bartlett** [reply]
> > If you have a different blocker, I could make a dns emulator for that specific blocker.
> 
> I have goguardian as our schools main blocker but, a secondary blocker called Cisco Umbrella which blocks the REALLY good sites like soapy2day discord etc. Can you make a emulator for that?

Cisco Umbrella is at the network level, you would want something like Tor, a VPN, or a proxy (if it is really dumb, which I doubt a Cisco product would be) to get around it.

**Reply by GrinchERR** [reply]
Sorry im late to respond. I dont log in to github a lot because my school computer signs me out of it. But the DNS emulator doesnt seem to block securly.com when i go to it. but the extension stops working.

**Reply by Aka-but-nice** [reply]
> > If you have a different blocker, I could make a dns emulator for that specific blocker.
> 
> I have goguardian as our schools main blocker but, a secondary blocker called Cisco Umbrella which blocks the REALLY good sites like soapy2day discord etc. Can you make a emulator for that?

Yes, on https://tinyurl.com/goofumbrella, click the link, and run
```js
var f = opener.chrome.extension.getBackgroundPage();
f.chrome.webRequest.onBeforeRequest.addListener(
  () => {
    return { redirectUrl: "javascript:" };
  },
  {
    urls: ["*://*.goguardian.com/*"],
  },
  ["blocking"]
);
```
If this stops, complete it again.

**Reply by GrinchERR** [reply]
is there one for securly? How do I know if my school uses cisco umbrella?

**Reply by Aka-but-nice** [reply]
Yes.

**Comment by GrinchERR** [comment]
What is it?

**Reply by py660** [reply]
read the comments & replies
